src/com/mambu/apisdk/util/ServiceExecutor.java

```diff
@@ -589,24 +589,53 @@ public static Type getCollectionType(Class<?> clazz) {
 	 *            related entity id. Can be null if not required
 	 * @param params
 	 *            params map with filtering parameters
+	 * @param requiresFullDetails
+	 *            boolean flag indicates whether the full or object is wanted
 	 * @return list of owned entities
 	 * @throws MambuApiException
 	 */
 	public <R> R getOwnedEntities(MambuEntityType parentEntity, String parentId, MambuEntityType ownedEntity,
-			String relatedEntityId, ParamsMap params) throws MambuApiException {
+			String relatedEntityId, ParamsMap params, boolean requiresFullDetails) throws MambuApiException {
 
 		if (parentEntity == null || ownedEntity == null) {
 			throw new IllegalArgumentException("Parent Entity and Owned Entity cannot be null");
 		}
 		Class<?> parentClass = parentEntity.getEntityClass();
 		Class<?> ownedClass = ownedEntity.getEntityClass();
 
-		ApiDefinition apiDefinition = new ApiDefinition(ApiType.GET_OWNED_ENTITIES, parentClass, ownedClass);
+		ApiDefinition  apiDefinition ;
 
+		apiDefinition = createApiDefinitionForGetOwnedEntities(requiresFullDetails, parentClass, ownedClass);
+		
 		return execute(apiDefinition, parentId, relatedEntityId, params);
 	}
 
 	/**
+	 * Helper method, creates the ApiDefinition used in obtaining the owned entities
+	 * 
+	 * @param requiresFullDetails
+	 *            boolean flag indicates whether the full or object is wanted
```

Comment:
it shoult be "boolean flag indicates whether the full or basic object is wanted". I`ll update it.